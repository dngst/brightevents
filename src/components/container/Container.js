import styled from "styled-components";

const Container = styled.div`
  text-align: ${(props) => (props.left ? "left" : "center")};
  margin: 4% auto;
  padding: 1% 0;
  width: ${(props) => (props.md ? "30%" : "55%")};

  button {
    width: ${(props) => (props.md ? "100%" : "max-content")};
  }

  @media (max-width: 768px) {
    padding: 10% 0;
    width: 95%;
    margin-top: 10%;
    p {
      font-size: 0.9rem;
    }
  }
`;
export default Container;
