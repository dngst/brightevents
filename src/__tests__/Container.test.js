import { render } from "@testing-library/react";
import Container from "components/container/Container";

it("works", () => {
  const { container } = render(<Container />);
  expect(container.firstChild).toMatchSnapshot();
});

it("applies default styles", () => {
  const { container } = render(<Container />);
  expect(container.firstChild).toHaveStyleRule("text-align", "center");
  expect(container.firstChild).toHaveStyleRule("margin", "4% auto");
  expect(container.firstChild).toHaveStyleRule("width", "55%");
  expect(container.firstChild).toHaveStyleRule("width", "95%", {
    media: "(max-width: 768px)",
  });
});

it("applies default styles according to passed props", () => {
  const { container } = render(<Container md />);
  expect(container.firstChild).toHaveStyleRule("width", "30%");
});
