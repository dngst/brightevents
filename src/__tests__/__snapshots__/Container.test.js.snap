// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
.c0 {
  text-align: center;
  margin: 4% auto;
  padding: 1% 0;
  width: 55%;
}

.c0 button {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

@media (max-width:768px) {
  .c0 {
    padding: 10% 0;
    width: 95%;
    margin-top: 10%;
  }

  .c0 p {
    font-size: 0.9rem;
  }
}

<div
  class="c0"
/>
`;
